File: app/src/main/java/br/pedroso/upcomingmovies/movieslist/di/MoviesPresenterModule.java
package br.pedroso.upcomingmovies.movieslist.di;

import br.pedroso.upcomingmovies.movieslist.MoviesListView;
import dagger.Module;
import dagger.Provides;

@Module
public class MoviesPresenterModule {
    private final MoviesListView view;

    public MoviesPresenterModule(MoviesListView view) {
        this.view = view;
    }

    @Provides
    MoviesListView providesMoviesView() {
        return view;
    }
}

File: app/src/main/kotlin/br/pedroso/upcomingmovies/movieslist/di/MoviesPresenterModule.kt
package br.pedroso.upcomingmovies.movieslist.di

import br.pedroso.upcomingmovies.movieslist.MoviesListView
import dagger.Module
import dagger.Provides

@Module
class MoviesPresenterModule(private val view: MoviesListView) {
    @Provides
    fun providesMoviesView(): MoviesListView {
        return view
    }
}
